src/sfizz/Smoothers.cpp

```diff
@@ -27,10 +27,25 @@ void Smoother::reset(float value)
 
 void Smoother::process(absl::Span<const float> input, absl::Span<float> output)
 {
-    if (smoothing)
+    CHECK_SPAN_SIZES(input, output);
+    if (input.size() == 0)
+        return;
+
+    const auto midValue = input[input.size() / 2];
+    const bool shortcut = (
+        input.front() == input.back()
+        && input.front() == midValue
```

Comment:
how valid is this as a guarantee that signal may be constant?
if we are smoothing a CC, it may well happen as a coincidence that 3 points of the CC stream are equal but the rest of signal isn't.

- beside even if the condition for shortcut is valid, what if we leave a state of filter smoothing which has not completed?
do we take a discontinuity.
- same after we leave shortcut state to go back to the filter, we accept the discontinuity there also?

Reply:
> how valid is this as a guarantee that signal may be constant?
> if we are smoothing a CC, it may well happen as a coincidence that 3 points of the CC stream are equal but the rest of signal isn't.

For sure we may miss these cases.

> 
>     * beside even if the condition for shortcut is valid, what if we leave a state of filter smoothing which has not completed? do we take a discontinuity?
> 
>     * same after we leave shortcut state to go back to the filter, we accept the discontinuity there also?

I think for these the condition
```c++
&& input.front() == current()
```
will be false.

Reply:
Maybe another way could be to check if there was an event in the block. If no event AND if the filter has finished smoothing to the expected value, then we can shortcut.

Reply:
Does this mean
- if there is no event ever received,the code is fast
- the moment the first cc arrives, the code is slow?

what is going to happen with floating point IIR? is a filter going to converge moderately "fast" to equal the target value? will it be guaranteed to converge ever?

Reply:
Anyway, I think shortcut can be deduced according to these conditions
- the cc stream is stationary ; defined as having no events in the current cycle
- the filter output is within some reasonable epsilon of the target

Reply:
> Anyway, I think shortcut can be deduced according to these conditions
> 
>     * the cc stream is stationary ; defined as having no events in the current cycle
> 
>     * the filter output is within some reasonable epsilon of the target

This is basically what I meant in the last reply. Let me try to provide an implementation.

Reply:
New tentative in dda4a94